pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -338,120 +344,180 @@ func TestInt_Tasks(t *testing.T) {
-		err := client.Tasks.Drop(ctx, sdk.NewDropTaskRequest(NonExistingSchemaObjectIdentifier))
-		assert.ErrorIs(t, err, sdk.ErrObjectNotExistOrAuthorized)
-	})
-
-	t.Run("alter task: set value and unset value", func(t *testing.T) {
-		task := createTask(t)
-		id := task.ID()
+		assertions.AssertThat(t, objectassert.TaskFromObject(t, task).
+			HasWarehouse(testClientHelper().Ids.WarehouseId().Name()).
+			HasSchedule("10 MINUTES").
+			HasConfig(`{"output_dir": "/temp/test_directory/", "learning_rate": 0.1}`).
+			HasAllowOverlappingExecution(true).
+			HasCondition(`SYSTEM$STREAM_HAS_DATA('MYSTREAM')`).
+			HasComment("some_comment").
+			HasTaskRelations(sdk.TaskRelations{}),
+		)
 
-		alterRequest := sdk.NewAlterTaskRequest(id).WithSet(sdk.NewTaskSetRequest().WithComment(sdk.String("new comment")).WithUserTaskTimeoutMs(sdk.Int(1000)))
-		err := client.Tasks.Alter(ctx, alterRequest)
+		err = client.Tasks.CreateOrAlter(ctx, sdk.NewCreateOrAlterTaskRequest(id, sql))
 		require.NoError(t, err)
 
 		alteredTask, err := client.Tasks.ShowByID(ctx, id)
 		require.NoError(t, err)
 
-		assert.Equal(t, "new comment", alteredTask.Comment)
+		assertions.AssertThat(t, objectassert.TaskFromObject(t, alteredTask).
+			HasWarehouse("").
+			HasSchedule("").
+			HasConfig("").
+			HasAllowOverlappingExecution(false).
+			HasCondition("").
+			HasComment("").
+			HasTaskRelations(sdk.TaskRelations{}),
+		)
 
-		alterRequest = sdk.NewAlterTaskRequest(id).WithUnset(sdk.NewTaskUnsetRequest().WithComment(sdk.Bool(true)).WithUserTaskTimeoutMs(sdk.Bool(true)))
-		err = client.Tasks.Alter(ctx, alterRequest)
-		require.NoError(t, err)
+		require.Equal(t, createdOn, alteredTask.CreatedOn)
+	})
 
-		alteredTask, err = client.Tasks.ShowByID(ctx, id)
+	t.Run("drop task: existing", func(t *testing.T) {
+		task, taskCleanup := testClientHelper().Task.Create(t)
+		t.Cleanup(taskCleanup)
+
+		err := client.Tasks.Drop(ctx, sdk.NewDropTaskRequest(task.ID()))
 		require.NoError(t, err)
 
-		assert.Equal(t, "", alteredTask.Comment)
+		_, err = client.Tasks.ShowByID(ctx, task.ID())
+		assert.ErrorIs(t, err, sdk.ErrObjectNotFound)
+	})
+
+	t.Run("drop task: non-existing", func(t *testing.T) {
+		err := client.Tasks.Drop(ctx, sdk.NewDropTaskRequest(NonExistingSchemaObjectIdentifier))
+		assert.ErrorIs(t, err, sdk.ErrObjectNotExistOrAuthorized)
+	})
+
+	t.Run("alter task: set value and unset value", func(t *testing.T) {
+		task, taskCleanup := testClientHelper().Task.Create(t)
+		t.Cleanup(taskCleanup)
+
+		err := client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(task.ID()).WithSet(*sdk.NewTaskSetRequest().
+			// TODO: Cannot set warehouse due to Snowflake error
+			// WithWarehouse(testClientHelper().Ids.WarehouseId()).
+			WithSchedule("10 MINUTE").
+			WithConfig(`$${"output_dir": "/temp/test_directory/", "learning_rate": 0.1}$$`).
+			WithAllowOverlappingExecution(true).
+			WithUserTaskTimeoutMs(1000).
+			WithSuspendTaskAfterNumFailures(100).
```

Comment:
What about session parameters?

Reply:
Added todo for now, I forgot about them and they turned out to be pretty big, I'll tackle them fully in the next pr.